lib/controller/CommissieVoorkeurenController.class.php

```diff
@@ -90,19 +91,36 @@ public function GET_overzicht($commissieId = null) {
 
 	public function nieuwecommissie() {
 		$model = new VoorkeurCommissie();
-		if ((new AddCommissieFormulier($model))->validate()) {
+		$form = (new AddCommissieFormulier($model));
+		if ($form->validate() && trim($model->naam) != '') {
 			$id = VoorkeurCommissieModel::instance()->create($model);
-			redirect("/commissievoorkeuren/overzicht/" . $id);
+			redirect('/commissievoorkeuren/overzicht/' . $id);
 		} else {
-			redirect("/commissievoorkeuren/");
+			setMelding('Naam van commissie mag niet leeg zijn', 2);
+			redirect('/commissievoorkeuren/');
 		}
 
 	}
 
 	public function nieuwecategorie() {
 		$model = new VoorkeurCommissieCategorie();
-		if ((new AddCategorieFormulier($model))->validate()) {
+		if ((new AddCategorieFormulier($model))->validate() && trim($model->naam) != '') {
```

Comment:
Met `RequiredTextField` is dit overbodig. Een `trim($value)` kan wel zinnig zijn in de validate van `TextField` of `InputField`, daar staat ie nu niet.

Als je een form instance maakt, deze validate en daarna naar de view teruggeeft, krijg je mooie error messages.

```
$form = new AddCategorieFormulier(new VoorkeurCommissieCategorie());
if ($form->validate()) {
  // Doe dingen
  redirect('/commissievoorkeuren/') // Prevent resubmit
}

$body = new CommissieVoorkeurenOverzicht(VoorkeurCommissieModel::instance()->getByCategorie(), null, $form);
$this->view = new CsrLayoutView($body);
```

En dan in `CommissieVoorkeurenOverzicht`

```
...
  protected $categorieFormulier;
  protected $commissieFormulier;

  public function __construct($categorien, $categorieFormulier = null, $commissieFormulier = null) {
		parent::__construct($categorien, "/commissievoorkeuren/");
    $this->categorieFormulier = $categorieFormulier ?? new AddCategorieFormulier(new VoorkeurCommissieCategorie());
    $this->commissieFormulier = $commissieFormulier ?? new AddCommissieFormulier(new VoorkeurCommissie());
	}
	public function view() {
		$this->smarty->assign("categorieFormulier", $this->categorieFormulier);
		$this->smarty->assign("commissieFormulier", $this->commissieFormulier);
		$this->smarty->assign("categorien", $this->model);
		$this->smarty->display("commissievoorkeuren/overzicht.tpl");
	}
...
```